core/src/main/java/com/datastax/oss/driver/internal/core/session/DefaultSession.java

```diff
@@ -393,6 +394,7 @@ private void afterInitialSchemaRefresh(CqlIdentifier keyspace) {
                   if (error != null) {
                     initFuture.completeExceptionally(error);
                   } else {
+                    notifyLifecycleListeners();
```

Comment:
Shouldn't the call come after `initFuture.complete(DefaultSession.this);`?

Reply:
It's really up to us to decide and I don't think it makes such a difference either way.

But given how I worded it in the javadocs, you're right, I should probably do it after.